src/utils/firebaseHelpers.ts

```diff
@@ -82,4 +98,62 @@ const getQuiz = async (quizId: string) => {
   }
 };
 
-export { createUserDocument, getUserDocument, getCategories, getQuizzes, getQuiz };
+const updateUserDocument = async (uid: string, currentScore: number) => {
+  try {
+    const userRef = doc(db, `users/${uid}`);
+    const userSnapshot = await getDoc(userRef);
+
+    if (userSnapshot.exists()) {
+      const { totalAttemptedQuiz, totalScore } = userSnapshot.data();
+
+      await updateDoc(userRef, {
+        totalAttemptedQuiz: totalAttemptedQuiz + 1,
+        totalScore: totalScore + currentScore,
+      });
+    }
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const addQuiz = async (quizData: {
+  quizName: string;
+  quizDescription: string;
+  quizCategory: string;
+  quizImage: string;
+}) => {
+  try {
+    const quizRef = collection(db, "quizzes");
+    const res = await addDoc(quizRef, { ...quizData, quizStatus: "available" });
+    return res.id;
+  } catch (err) {
+    return "something went wrong";
```

Comment:
In ```addQuestion``` function, youre using ```getErrorMessage``` to get the error message, you can use that function here as well